server/src-lib/Hasura/Server/Config.hs

```diff
@@ -10,13 +10,16 @@ import           Hasura.Prelude
 import           Hasura.Server.Auth
 import           Hasura.Server.Auth.JWT
 import           Hasura.Server.Version                    (HasVersion, Version, currentVersion)
+import           Hasura.RQL.Types.Error                   (encodeJSONPath)
+import           Data.Text                                (pack)
 
 import qualified Hasura.GraphQL.Execute.LiveQuery.Options as LQ
 
 data JWTInfo
   = JWTInfo
-  { jwtiClaimsNamespace :: !Text
-  , jwtiClaimsFormat    :: !JWTClaimsFormat
+  { jwtiClaimsNamespace     :: !(Maybe Text)
+  , jwtiClaimsNamespacePath :: !(Maybe Text)
```

Comment:
Both claims namespace and namespace path are orthogonal and only one should present at a time. Right? Then you could model these into a sum type.

Reply:
Both of the fields can be set at the same time, if both are set then we prefer the `claims_namespace_path` value. Initially, I thought we can display both the values but then I changed the logic of the claims namespace which will discard the `claims_namespace` value if `claims_namespace_path` is present

Reply:
I am kinda leaning towards the behaviour that we throw an error if both are set at the same time.

Reply:
If at all someone has a use-case where they want such fall-back behaviour, we can introduce this.

The main reason is to keep the config tight and obvious in reasoning.